modules/tinymce/src/core/main/ts/selection/SplitRange.ts

```diff
@@ -13,30 +13,32 @@ const split = (rng: RangeLikeObject): RangeLikeObject => {
 
   // Handle single text node
   if (startContainer === endContainer && NodeType.isText(startContainer)) {
-    if (startOffset > 0 && startOffset < startContainer.nodeValue.length) {
+    if (startOffset > 0 && startOffset < startContainer.data.length) {
       endContainer = splitText(startContainer, startOffset);
-      startContainer = endContainer.previousSibling;
+      startContainer = endContainer.previousSibling as Text;
 
       if (endOffset > startOffset) {
         endOffset = endOffset - startOffset;
-        startContainer = endContainer = splitText(endContainer as Text, endOffset).previousSibling;
-        endOffset = endContainer.nodeValue.length;
+        const newContainer = splitText(endContainer as Text, endOffset).previousSibling as Text;
```

Comment:
Extracted this out to a separate variable so we can assert the type, since TS can't track that the split nodes previousSibling will definitely be a Text node here.